wrapper/src/main/java/software/amazon/jdbc/HikariPooledConnectionProvider.java

```diff
@@ -98,6 +104,10 @@ public HikariPooledConnectionProvider(
       HikariPoolConfigurator hikariPoolConfigurator, HikariPoolMapping mapping) {
     this.poolConfigurator = hikariPoolConfigurator;
     this.poolMapping = mapping;
+    final LeastConnectionsHostSelector hostSelector =
+        (LeastConnectionsHostSelector) acceptedStrategies
```

Comment:
That's dangerous casting. Need to refactor the code a bit. It seems that LeastConnectionsHostSelector needs access to connection provider internals. I'd recommend initialize an instance of LeastConnectionsHostSelector in connection provider constructor and pass a reference to itself.